language/compiler/bytecode-source-map/src/source_map.rs

```diff
@@ -71,6 +71,9 @@ pub struct SourceMap<Location: Clone + Eq> {
 
     // A mapping of FunctionDefinitionIndex to the soure map for that function.
     function_map: BTreeMap<TableIndex, FunctionSourceMap<Location>>,
+
+    // A mapping of ConstantPoolIndex to a vector of constant names.
+    pub constant_map: BTreeMap<TableIndex, Vec<ConstantName>>,
```

Comment:
Probably a stupid question, but this seems to imply that  a constant at a table index can have multiple names, which doesn't seem right. Shouldn't it be something like `BTreeMap<TableIndex, ConstantName>`?

Reply:
There can indeed be multiple names. If we have two constants with the same value, they will be mapped to the same index in the constant pool.